logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/mlsmigration/MLSMigrator.kt

```diff
@@ -83,13 +94,32 @@ internal class MLSMigratorImpl(
 
     private suspend fun finalise(conversationId: ConversationId): Either<CoreFailure, Unit> {
         kaliumLogger.i("finalising ${conversationId.toLogString()} to mls")
-        return conversationRepository.updateProtocol(conversationId, Conversation.Protocol.MLS)
-            .flatMapLeft {
-                kaliumLogger.w("failed to migrate ${conversationId.toLogString()} to mls: $it")
+        return conversationRepository.updateProtocol(conversationId, Protocol.MLS)
+            .fold({ failure ->
+                kaliumLogger.w("failed to migrate ${conversationId.toLogString()} to mls: $failure")
                 Either.Right(Unit)
-            }
+            }, { updated ->
+                if (updated) {
+                    systemMessageBuilder.insertProtocolChangedSystemMessage(
+                        conversationProtocolEvent(conversationId, Protocol.MLS)
+                    )
+                }
+                Either.Right(Unit)
+            })
     }
 
+    private fun conversationProtocolEvent(
```

Comment:
maybe 
```suggestion
    private fun handleConversationProtocolEvent(
```

or 
```suggestion
    private fun onConversationProtocolEvent(
```

Reply:
I'm handling the event I'm constructing an event but I think I'll change the system message builder to not take an event as a parameter.

Reply:
Removed this method completely.